nodes/test/mcp_client/test_sse_redirect.py

```
@@ -0,0 +1,135 @@
+
+
+class TestEndpointRedirectValidation(unittest.TestCase):
+    """Ensure the endpoint event validates same-origin."""
+
+    # ---- Cases that MUST be accepted (same origin) ----
+
+    def test_relative_url_accepted(self):
+        """A normal relative path should be resolved and accepted."""
+        client = _make_client('https://legit-server.com/sse')
+        client._handle_sse_event(event='endpoint', data='/messages?session=abc123')
+        self.assertEqual(
```

Ruff (0.15.7)

[warning] Use a regular `assert` instead of unittest-style `assertEqual`

Replace `assertEqual(...)` with `assert ...`

(PT009)